### PR TITLE
Remove null checks from SdkSpanBuilder

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/AttributesMap.java [hunk 1]
```diff
@@ -27,9 +27,6 @@
   }
 
   <T> void put(AttributeKey<T> key, T value) {
-    if (key == null || key.getKey() == null || value == null) {
```
Comment (anuraaga) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/AttributesMap.java:30]: Checked by all callers, this is an internal class

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java [hunk 2]
```diff
@@ -251,7 +251,7 @@ Clock getClock() {
 
   @Override
   public <T> ReadWriteSpan setAttribute(AttributeKey<T> key, T value) {
-    if (key == null || key.getKey() == null || key.getKey().length() == 0 || value == null) {
```
Comment (anuraaga) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java:254]: Removing the unnecessary key null check (null key mapped to empty string now) is originally what had me find the other ones we never removed

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java [hunk 3]
```diff
@@ -109,7 +112,6 @@ public SpanBuilder addLink(SpanContext spanContext, Attributes attributes) {
   }
 
   private void addLink(LinkData link) {
-    Objects.requireNonNull(link, "link");
```
Comment (anuraaga) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java:112]: Private method

**Review by jkwatson — APPROVED** [review]
Thanks!

**Review by carlosalberto — APPROVED** [review]

**Pull request merged**